cmd/geth/chaincmd.go

```diff
@@ -793,3 +834,269 @@ func parseRange(s string) (start uint64, end uint64, ok bool) {
+				Nonce:    pbTx.Nonce,
+				To:       func() *common.Address { addr := common.BytesToAddress(pbTx.To); return &addr }(),
+				Value:    pbTx.Value.Native(),
+				Gas:      pbTx.GasLimit,
+				GasPrice: pbTx.GasPrice.Native(),
+				Data:     pbTx.Input,
+				V:        new(big.Int).SetBytes(pbTx.V),
+				R:        new(big.Int).SetBytes(pbTx.R),
+				S:        new(big.Int).SetBytes(pbTx.S),
+			})
+		} else {
+			tx = types.NewTx(&types.LegacyTx{
+				Nonce:    pbTx.Nonce,
+				To:       nil,
+				Value:    pbTx.Value.Native(),
+				Gas:      pbTx.GasLimit,
+				GasPrice: pbTx.GasPrice.Native(),
+				Data:     pbTx.Input,
+				V:        new(big.Int).SetBytes(pbTx.V),
+				R:        new(big.Int).SetBytes(pbTx.R),
+				S:        new(big.Int).SetBytes(pbTx.S),
+			})
+		}
+
+		if tx != nil {
+			txs = append(txs, tx)
+		}
+	}
+
+	// Convert uncles
+	var uncles []*types.Header
+	for _, pbUncle := range pbBlock.Uncles {
+		if pbUncle == nil {
+			continue
+		}
+		uncle := &types.Header{
+			ParentHash:  common.Hash(pbUncle.ParentHash),
+			UncleHash:   common.Hash(pbUncle.UncleHash),
+			Coinbase:    common.Address(pbUncle.Coinbase),
+			Root:        common.Hash(pbUncle.StateRoot),
+			TxHash:      common.Hash(pbUncle.TransactionsRoot),
+			ReceiptHash: common.Hash(pbUncle.ReceiptRoot),
+			Bloom:       types.BytesToBloom(pbUncle.LogsBloom),
+			Difficulty:  pbUncle.Difficulty.Native(),
+			Number:      big.NewInt(int64(pbUncle.Number)),
+			GasLimit:    pbUncle.GasLimit,
+			GasUsed:     pbUncle.GasUsed,
+			Time:        uint64(pbUncle.Timestamp.Seconds),
+			Extra:       pbUncle.ExtraData,
+			MixDigest:   common.Hash(pbUncle.MixHash),
+			Nonce:       types.EncodeNonce(pbUncle.Nonce),
+		}
+		if pbUncle.BaseFeePerGas != nil {
+			uncle.BaseFee = pbUncle.BaseFeePerGas.Native()
+		}
+
+		if pbUncle.WithdrawalsRoot != nil {
+			header.WithdrawalsHash = (*common.Hash)(pbBlock.Header.WithdrawalsRoot)
+		}
+
+		if pbUncle.BlobGasUsed != nil {
+			header.BlobGasUsed = pbBlock.Header.BlobGasUsed
+		}
+
+		if pbUncle.ExcessBlobGas != nil {
+			header.ExcessBlobGas = pbBlock.Header.ExcessBlobGas
+		}
+
+		if pbUncle.ParentBeaconRoot != nil {
+			header.ParentBeaconRoot = (*common.Hash)(pbBlock.Header.ParentBeaconRoot)
+		}
+
+		if pbUncle.RequestsHash != nil {
+			header.ParentBeaconRoot = (*common.Hash)(pbBlock.Header.RequestsHash)
+		}
+
+		uncles = append(uncles, uncle)
+	}
+
+	body := &types.Body{
+		Transactions: txs,
+		Uncles:       uncles,
+	}
+
+	// TODO: Receipts and Hasher
+
+	// Create block
+	return types.NewBlock(header, body, nil, nil), nil
+}
+
+// writeBatch writes a batch of blocks to an RLP file
+func writeBatch(blocks []*types.Block, outputPrefix string, batchNum int) error {
+	dir := "rlp-data"
+	if err := os.MkdirAll(dir, 0o755); err != nil {
+		return fmt.Errorf("failed to create directory %s: %w", dir, err)
+	}
+
+	var filename string
+	if batchNum == 0 {
+		filename = fmt.Sprintf("%s/%s.rlp", dir, outputPrefix)
```

Comment:
`filepath.Join` would be better than the litteral `/`, sometimes a user pops up using Windows or something...